src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -201,131 +201,137 @@ protected void doCheckExistingTable(NosqlEntityInformation<?, ?> entityInformati
+                errors.add(msg);
+            }
 
-                // check identity same
-                FieldValue identity = tableSchema.get(identityField);
-                if (identity != null && !entityInformation.isAutoGeneratedId()) {
-                    errors.add("Identity information mismatch.");
+            // check identity same
+            FieldValue identity = tableSchema.get(identityField);
+            if (identity != null && !entityInformation.isAutoGeneratedId()) {
+                errors.add("Identity information mismatch.");
 
-                } else if (identity == null && entityInformation.isAutoGeneratedId() &&
-                        entityInformation.getIdNosqlType() != FieldValue.Type.STRING) {
-                    errors.add("Identity information mismatch.");
-                }
+            } else if (identity == null && entityInformation.isAutoGeneratedId() &&
+                    entityInformation.getIdNosqlType() != FieldValue.Type.STRING) {
+                errors.add("Identity information mismatch.");
+            }
 
-                // TTL warning
-                FieldValue ttlValue = tableSchema.get(ttlField);
-                TimeToLive ttl = entityInformation.getTtl();
-                // TTL is present in database but not in the entity
-                if (ttlValue != null && ttl != null &&
-                        !ttl.toString().equalsIgnoreCase(ttlValue.getString())) {
-                    LOG.warn("TTL of the table in database is different from " +
-                            "the TTL of the entity " +
-                            entityInformation.getJavaType().getName());
-                } else if (ttlValue == null && ttl != null && ttl.getValue() != 0) {
-                    // TTL is present in entity but not in the database
-                    LOG.warn("TTL of the table in database is different from " +
-                            "the TTL of the entity " +
-                            entityInformation.getJavaType().getName());
-                }
+            // TTL warning
+            FieldValue ttlValue = tableSchema.get(ttlField);
+            TimeToLive ttl = entityInformation.getTtl();
+            // TTL is present in database but not in the entity
+            if (ttlValue != null && ttl != null &&
+                    !ttl.toString().equalsIgnoreCase(ttlValue.getString())) {
+                LOG.warn("TTL of the table in database is different from " +
+                        "the TTL of the entity " +
+                        entityInformation.getJavaType().getName());
+            } else if (ttlValue == null && ttl != null && ttl.getValue() != 0) {
+                // TTL is present in entity but not in the database
+                LOG.warn("TTL of the table in database is different from " +
+                        "the TTL of the entity " +
+                        entityInformation.getJavaType().getName());
+            }
+        } catch (NullPointerException npe) {
```

Comment:
Is this necessary? Isn't there a better way to detect this before throwing the NPE?